File: frontend/src/apis/agdevx-cart-api/models/inventory-item.ts
// ABOUTME: InventoryItem interface representing household-shared or user-private items
// ABOUTME: Household items are visible to all household members, user items are private to owner

export interface InventoryItem {
  id: string;
  name: string;
  defaultStoreId: string | null;
  notes: string | null;
  ownerUserId: string | null;
  householdId: string | null;
  createdBy: string | null;
  createdDate: string;
  modifiedBy: string | null;
  modifiedDate: string | null;
}

File: backend/AGDevX.Cart.Data/Models/InventoryItem.cs
// ABOUTME: InventoryItem model representing household-shared or user-private items
// ABOUTME: Household items are visible to all household members, user items are private to owner
namespace AGDevX.Cart.Data.Models;

public class InventoryItem : BaseEntity
{
    public required string Name { get; set; }

    //== Optional default store where this item is typically found
    public Guid? DefaultStoreId { get; set; }
    public Store? DefaultStore { get; set; }

    public string? Notes { get; set; }

    //== User-scoped item (private to owner)
    public Guid? OwnerUserId { get; set; }
    public User? OwnerUser { get; set; }

    //== Household-scoped item (shared across household)
    public Guid? HouseholdId { get; set; }
    public Household? Household { get; set; }
}
